Lib/sndhdr.py

```diff
@@ -184,14 +193,19 @@ def test_sndt(h, f):
         rate = get_short_le(h[20:22])
         return 'sndt', rate, 1, nsamples, 8
 
+
 tests.append(test_sndt)
 
 
 def test_sndr(h, f):
-    if h.startswith(b'\0\0'):
-        rate = get_short_le(h[2:4])
-        if 4000 <= rate <= 25000:
-            return 'sndr', rate, 1, -1, 8
+    try:
+        if h.startswith(b'\0\0'):
+            rate = get_short_le(h[2:4])
+            if 4000 <= rate <= 25000:
+                return 'sndr', rate, 1, -1, 8
+    except IndexError:
```

Comment:
What is the IndexError? Is it when h is empty?

Reply:
in that case, check len(h) instead of *expecting* an IndexError.

Reply:
lol this is a good point. I'll check the length. thanks!

Reply:
if the length is less than 2, what should `get_short_le()` do? just return `None`? or raise an Error? 

Whatever change that is, should it also apply to the other `get_x_x()` subroutines?

Reply:
Catching an IndexError and checking `len(h)` look equally good to me.

Reply:
I'll go for checking `len()`.